scripts/rv_expression_association/get_rv_outliers.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import click
+import logging
+import hail as hl
+import pandas as pd
+from cpg_utils.hail_batch import reference_path
+from cpg_utils.hail_batch import output_path, init_batch
+from cloudpathlib import AnyPath
+
+# use logging to print statements, display at info level
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+# get OneK1K sample ID and gene name as arguments using click
+@click.command()
+@click.option('--onek1k-id', required=True)
+@click.option('--gene-name', required=True)
+@click.option('--sample-mapping-file', required=True)
+@click.option('--gene-file', required=True)
+@click.option(
+    '--window-size',
+    required=True,
+    help='the size of the flanking region, to be added twice, on either side of the gene body',
+)
+def main(
+    onek1k_id: str,
+    gene_name: str,
+    sample_mapping_file: str,  # 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/OneK1K_CPG_IDs.tsv'
+    gene_file: str,  # e.g., 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/gene_location_files/GRCh38_geneloc_chr1.tsv'
+    window_size: int,
+):
+    """for a given individual,
+    and gene for which that individual is an expression outlier
+    get all variants within a given window,
+
+    then filter for variants that:
+    - are alt (0/1 or 1/1) for that individual
+    - are biallelic SNVs
+    - have regulatory consequences based on VEP annotations
+
+    also annotate variants with the following:
+    - CADD score
+    - MAF within the OneK1K sample
+    - MAF in gnomad
+
+    create the following table:
+    donor ID | gene ID | variant ID | position | CADD | MAF (OneK1K) | MAF (gnomad) | ..
+    """
+
+    # file matching OneK1K IDs to CPG (internal) and TOB (external) IDs
+    sample_key_df = pd.read_csv(AnyPath(sample_mapping_file), sep='\t', index_col=0)
```

Comment:
@illusional We don't have both of these external IDs in Metamist, I assume?

Reply:
I don't think we have OneK1K IDs, or at least I don't remember seeing them. W would want to address https://github.com/populationgenomics/sample-metadata/issues/139 before we ingested them. But seems relatively important.